nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/record/sink/script/ScriptedRecordSink.java

```diff
@@ -217,26 +211,31 @@ public void onEnabled(final ConfigurationContext context) {
         super.onEnabled(context);
 
         // Call an non-interface method onEnabled(context), to allow a scripted RecordSinkService the chance to set up as necessary
-        final Invocable invocable = (Invocable) scriptEngine;
-        if (configurationContext != null) {
-            try {
-                // Get the actual object from the script engine, versus the proxy stored in RecordSinkService. The object may have additional methods,
-                // where RecordSinkService is a proxied interface
-                final Object obj = scriptEngine.get("recordSink");
-                if (obj != null) {
-                    try {
-                        invocable.invokeMethod(obj, "onEnabled", context);
-                    } catch (final NoSuchMethodException nsme) {
-                        if (getLogger().isDebugEnabled()) {
-                            getLogger().debug("Configured script RecordSinkService does not contain an onEnabled() method.");
+        if (scriptRunner != null) {
+            final ScriptEngine scriptEngine = scriptRunner.getScriptEngine();
+            final Invocable invocable = (Invocable) scriptEngine;
```

Comment:
In the reloadScript() method there is a check:
`if (scriptEngine instanceof Invocable)`
Is it not needed here as well?

Reply:
Yeah there might be a couple spots where the check should be done, probably a copy-paste error, I'll take a look. Good catch!

I think the script engines we include that can actually subclass a Processor or other component are all Invocable, but we should check here defensively.

Reply:
I decided to update the `createResources()` method to take a boolean (defaulting to true) to indicate whether the supported script engines are required to be Invocable. I'll change ExecuteScript to call it with false, and the rest will only allow a choice of an Invocable engine, so the individual checks should be unnecessary.